packages/terra-functional-testing/tests/jest/commands/expect/toMatchReference.test.js

```diff
@@ -299,12 +326,69 @@ describe('toMatchReference', () => {
       isSameDimensions: false,
       misMatchPercentage: 0.10,
     };
+    jest.spyOn(fs, 'existsSync').mockImplementation(() => true);
 
-    const result = toMatchReference(receivedScreenshot);
-    const expectedMessage = 'Expected the screenshot to match the reference screenshot, but received a screenshot with different dimensions.\nExpected the screenshot to be within the mismatch tolerance, but received a mismatch difference of 0.1%.\nScreenshot has changed and needs to be reviewed.';
+    const result = toMatchReference(receivedScreenshot, 'TestName');
+    const expectedMessage = 'Expected the screenshot to match the reference screenshot, but received a screenshot with different dimensions.\nExpected the screenshot to be within the mismatch tolerance, but received a mismatch difference of 0.1%.';
+
+    expect(result.pass).toBe(true);
+    expect(result.message()).toEqual(expectedMessage);
+    expect(mockInfo).toHaveBeenCalledWith('Test: \'TestName\' has a mismatch difference of 0.1% and needs to be reviewed.');
+  });
+
+  it('should create the output directory and ignored-mismatch file if useRemoteReferenceScreenshots is true, a mismatch happened, and it is a pull request', () => {
+    global.Terra = {
+      serviceOptions: {
+        buildBranch: BUILD_BRANCH.master,
+        buildType: BUILD_TYPE.branchEventCause,
+        useRemoteReferenceScreenshots: true,
+      },
+    };
+    const receivedScreenshot = {
+      isSameDimensions: false,
+      misMatchPercentage: 0.10,
+    };
+    jest.spyOn(fs, 'existsSync').mockImplementationOnce(() => false);
+    jest.spyOn(fs, 'mkdirSync').mockImplementationOnce(() => false);
+    jest.spyOn(fs, 'writeFileSync').mockImplementationOnce(() => false);
```

Comment:
We can also use the shorthand [mockFn.mockReturnValueOnce](https://jestjs.io/docs/mock-function-api#mockfnmockreturnvalueoncevalue)

Reply:
Fixed here: https://github.com/cerner/terra-toolkit/pull/839/commits/e3f1c358ea6ee77af9b877d072b9ca9f77df5871#diff-4fc55674149c158d6de22cac25bcc78653afd13a67eb1270cdd05c57b4abe9ebR350-R352